plugins/PowerManagement/PowerManagementLinux.py

```diff
@@ -0,0 +1,70 @@
+'''
+Created on 10.01.2017
+
+@author: thopiekar
+'''
+
+import dbus
+
+from UM.Extension import Extension
+
+from UM.Logger import Logger
+
+class PowerManagement(Extension):
+    def __init__(self):
+        super().__init__()
+                
+        self.suspendInhibition = None
+        
+        __init_api_list__ = [self.__init_api_solid__,
+                             self.__init_api_freedesktop__,
+                             ]
+        
+        while __init_api_list__:
+            try:
+                __init_api_list__[0]()
+                return
+            except:
+                del __init_api_list__[0]
+        
+        Logger.log("e", "We were not able to provide any power management provider!")
+        return
+    
+    def __init_api_solid__(self):
+            self.solid_session = None
+            devobj = dbus.SessionBus().get_object("org.kde.kded5",
```

Comment:
But the freedesktop interface doesn't allow to pass the application icon, afaik.
I couldn't find a way to pass the icon at the moment, but I'm planning to find it out later.